src/assets/AssetsContractController.test.ts

```diff
@@ -21,29 +25,40 @@ const ERC1155_ID =
 
 const TEST_ACCOUNT_PUBLIC_ADDRESS =
   '0x5a3CA5cD63807Ce5e4d7841AB32Ce6B6d9BbBa2D';
-describe('AssetsContractController', () => {
-  let assetsContract: AssetsContractController;
-  let preferences: PreferencesController;
-  let network: NetworkController;
-
-  beforeEach(() => {
-    preferences = new PreferencesController();
-    network = new NetworkController();
-    assetsContract = new AssetsContractController({
-      onPreferencesStateChange: (listener) => preferences.subscribe(listener),
-      onNetworkStateChange: (listener) => network.subscribe(listener),
+
+const setupControllers = () => {
+  const messenger: NetworkControllerMessenger =
+    new ControllerMessenger().getRestricted({
+      name: 'NetworkController',
+      allowedEvents: ['NetworkController:stateChange'],
+      allowedActions: [],
     });
+  const network = new NetworkController({
+    messenger,
+  });
+  const preferences = new PreferencesController();
+  const assetsContract = new AssetsContractController({
+    onPreferencesStateChange: (listener) => preferences.subscribe(listener),
+    onNetworkStateChange: (listener) =>
+      messenger.subscribe('NetworkController:stateChange', listener),
   });
 
+  return { messenger, network, preferences, assetsContract };
+};
+
+describe('AssetsContractController', () => {
   it('should set default config', () => {
+    const { assetsContract, messenger } = setupControllers();
     expect(assetsContract.config).toStrictEqual({
       chainId: SupportedTokenDetectionNetworks.mainnet,
       ipfsGateway: IPFS_DEFAULT_GATEWAY_URL,
       provider: undefined,
     });
+    messenger.clearEventSubscriptions('NetworkController:stateChange');
```

Comment:
One pattern I've been following in other projects recently to handle steps that you'd normally put into an `afterEach` is to have your setup function take a function. I like to rename the function `withXXXX` to indicate that it doesn't return the controllers (or test data) itself. For instance you could have:

```
function withControllers(fn): Controllers {
  // ...instantiate controllers...

  try {
    fn({ messenger, network, preferences, assetsContract });
  } finally {
    messenger.clearEventSubscriptions('NetworkController:stateChange');
  }
}
```

The reason `withControllers` takes a function is because that would now represent the test. Here's how you'd update an existing test for instance:

```
  it('should update the ipfsGateWay config value when this value is changed in the preferences controller', () => {
    withControllers(({ assetsContract, messenger, preferences }) => {
      expect(assetsContract.config).toStrictEqual({
        chainId: SupportedTokenDetectionNetworks.mainnet,
        ipfsGateway: IPFS_DEFAULT_GATEWAY_URL,
        provider: undefined,
      });
      preferences.setIpfsGateway('newIPFSGateWay');
      expect(assetsContract.config).toStrictEqual({
        ipfsGateway: 'newIPFSGateWay',
        chainId: SupportedTokenDetectionNetworks.mainnet,
        provider: undefined,
      });
    });
  });
```

I realize you've spent a lot of time on these tests already, so this isn't a request for this PR, but could give you ideas for a future PR.